src/main/java/org/icgc/argo/workflow_management/graphql/GatekeeperDataFetchers.java

```diff
@@ -0,0 +1,88 @@
+package org.icgc.argo.workflow_management.graphql;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static org.icgc.argo.workflow_management.util.JacksonUtils.convertValue;
+
+import com.google.common.collect.ImmutableList;
+import graphql.schema.DataFetcher;
+import java.util.List;
+import lombok.val;
+import org.icgc.argo.workflow_management.gatekeeper.model.ActiveRun;
+import org.icgc.argo.workflow_management.gatekeeper.service.GateKeeperService;
+import org.icgc.argo.workflow_management.graphql.model.GqlPage;
+import org.icgc.argo.workflow_management.graphql.model.GqlSort;
+import org.icgc.argo.workflow_management.graphql.model.SearchResult;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.domain.Example;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GatekeeperDataFetchers {
+
+  @Bean
+  @Profile("gatekeeper")
+  @Qualifier("activeRunsDataFetcher")
+  public DataFetcher getActiveRunsDataFetcher(GateKeeperService gateKeeperService) {
+    return environment -> {
+      val args = environment.getArguments();
+
+      val sortsBuilder = ImmutableList.<GqlSort>builder();
+      GqlPage page = null;
+      ActiveRun activeRun = null;
+      if (args != null) {
+        if (args.get("example") != null)
+          activeRun = convertValue(args.get("example"), ActiveRun.class);
+        if (args.get("page") != null) page = convertValue(args.get("page"), GqlPage.class);
+        if (args.get("sorts") != null) {
+          val rawSorts = (List<Object>) args.get("sorts");
+          sortsBuilder.addAll(
+              rawSorts.stream()
+                  .map(sort -> convertValue(sort, GqlSort.class))
+                  .collect(toUnmodifiableList()));
+        }
```

Comment:
This may be overkill, might also spark a different approach to this ... but could do something like this instead:

Have an inner class that captures your query object:

```
@Builder
@Data
class GqlQueryArgs {
  private ActiveRun activeRun;
  private GqlPage page;
  private ImmutableList.Builder<GqlSort> sortsBuilder;
}
```

Build it something like this:

```
val gqlQueryArgsBuilder = GqlQueryArgs.builder();

environment.getArguments().entrySet().stream()
    .filter(entry -> entry.getValue() != null)
    .forEach(
        entry -> {
          switch (entry.getKey()) {
            case "example":
              gqlQueryArgsBuilder.activeRun(convertValue(entry.getValue(), ActiveRun.class));
            case "page":
              gqlQueryArgsBuilder.page(convertValue(entry.getValue(), GqlPage.class));
            case "sorts":
              gqlQueryArgsBuilder.sortsBuilder(ImmutableList.<GqlSort>builder().addAll(
                      ((List<Object>) entry.getValue())
                              .stream()
                              .map(sort -> convertValue(sort, GqlSort.class))
                              .collect(toUnmodifiableList())));
          }
        });

val gqlQueryArgs = gqlQueryArgsBuilder.build();
```

There are probably "better" ways of doing that but just an idea